### PR TITLE
fetch APK keyring via https

### LINKED ISSUE #27: add support for downloading keyring keys
For distributions other than Alpine v3.14 through edge, due to ongoing evolution of the keyring, the default keyring is inadequate.  Thus, it's important to be able to bootstrap the keyring with keys stored outside the apko Docker image.

Accordingly, it would be nice to be able to download the keys necessary to bootstrap the keyring.  This would make it easier to use apko with distributions such as OpenWrt and Adelie too.

### PR DESCRIPTION
**Pull request by kaniini**
Add support for fetching an APK keyring via https.  Closes #27.

### CODE REVIEW

File: pkg/build/apk.go [hunk 1]
```diff
@@ -85,9 +89,31 @@ func (bc *Context) InitApkKeyring() (err error) {
 	for _, element := range keyFiles {
 		log.Printf("installing key %v", element)
 
-		data, err := os.ReadFile(element)
+		asURI := uri.New(element)
+		asURL, err := url.Parse(string(asURI))
```
Comment (mattmoor) [inline pkg/build/apk.go:93]: Why are you parsing it twice?

File: pkg/build/apk.go [hunk 2]
```diff
@@ -85,9 +89,31 @@ func (bc *Context) InitApkKeyring() (err error) {
...
+		case "file":
+			data, err = os.ReadFile(element)
+			if err != nil {
+				return errors.Wrap(err, "failed to read apk key")
+			}
+		case "https":
+			resp, err := http.Get(asURL.String())
+			if err != nil {
+				return errors.Wrap(err, "failed to fetch apk key")
+			}
```
Comment (mattmoor) [inline pkg/build/apk.go:109]: Go doesn't return an `error` when the status code is not `2xx`, so you should also check `resp.StatusCode` and `defer resp.Body.Close()`

File: pkg/build/apk.go (hunk 1 above)
Comment (kaniini) [reply pkg/build/apk.go:93]: The first time is to normalize the input into a URI, which can then be used with `net/url`.

File: pkg/build/apk.go (hunk 2 above)
Comment (kaniini) [reply pkg/build/apk.go:112]: Handled

File: pkg/build/apk.go (hunk 1 above)
Comment (kaniini) [reply pkg/build/apk.go:96]: I added a comment explaining it.

**Review by mattmoor — APPROVED** [review]

**Pull request merged**